src/main/java/org/example/learningsystem/LearningSystemApplication.java

```diff
@@ -2,9 +2,13 @@
 
 import org.springframework.boot.SpringApplication;
 import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.cache.annotation.EnableCaching;
+import org.springframework.retry.annotation.EnableRetry;
 import org.springframework.scheduling.annotation.EnableScheduling;
 
 @SpringBootApplication
+@EnableCaching
+@EnableRetry
 @EnableScheduling
```

Comment:
In case you have dedicated confiuration class for cache, retry, scheduling and so on, it's better to put annotations like `@EnableCaching`, `@EnableScheduling` and etc on these classes. It helps follow package by feature approach (if then you will decide to delete cache feature, you can just remove full package and no other classes/packages are affected)